src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -56,19 +72,47 @@ public Command parseCommand(String input) {
             return new IncorrectCommand(MESSAGE_INVALID_COMMAND_FORMAT);
         }
         String commandWord = matcher.group(COMMAND_WORD_GROUP).toLowerCase().trim();
+        String commandFlag = matcher.group(COMMAND_FLAG_GROUP).toLowerCase();
         String parameters = matcher.group(PARAMETERS_GROUP);
 
         try {
             switch (commandWord){
```

Comment:
This method seems a bit long, maybe do more abstraction

Reply:
i dont think i can abstract this switch statement due to the fact that it returns different methods based on the command_word_****? Any advice?